### PR TITLE
Test, Signup, Base forms

### LINKED ISSUE #25: App doesn't have a signup form.
A user must be able to signup for Potato Mail if they don't have an account previously.

### PR DESCRIPTION
**Pull request by jlc1541**
Contains a few labels to inform the reader on what information should be entered and where they should place it on the signup form.
Base.html has been changed to move the hyperlinks to the center of the screen and remove decorations.
Test.html has changed the words to "Welcome to PotatoMail" and was given a title attribute when hovering over it.

### CODE REVIEW

File: app/templates/base.html [hunk 1]
```diff
@@ -1,45 +1,23 @@
...
+        {% with messages = get_flashed_messages() %}
+        {% if messages %}
+        <ul>
+            {% for message in messages %}
+            <li>{{ message }}</li}
+            {% endfor %}
+        </ul>
+        {% endif %}
+        {% endwith %}
+        {% block content %}{% endblock %}
```
Comment (haleyhagler227) [inline app/templates/base.html]: for compatibility with the rest of the app change content to app_content

File: app/templates/base.html [hunk 2]
```diff
@@ -1,45 +1,23 @@
...
+<html>
+    <head>
+    </head>
+    <body>
+    
+        <div style="margin: 50px;text-align: center; font-size: 30px;">
+                <a href="{{url_for('test') }}" style="text-decoration:none">Home</a>
+                <a href="{{url_for('login') }}"style="text-decoration:none">Login</a>
+                <a href="{{url_for('SignUp') }}"style="text-decoration:none">SignUp</a>
+        </div>
```
Comment (haleyhagler227) [inline app/templates/base.html]: Change this back to block nav bar for compatibility with other parts of the app

File: app/templates/base.html [hunk 3]
```diff
@@ -1,45 +1,23 @@
...
-            <div class="nav navbar-nav">
-                <li><a href="{{url_for('test') }}">Home</a></li>
-                <li><a href="{{url_for('login') }}">Login</a></li>
-                <li><a href="{{url_for('index') }}">Inbox</a></li>
-                
-                
-            </div>
-        </div>
-    </nav>
-{% endblock %}
```
Comment (haleyhagler227) [inline app/templates/base.html]: This is the nav bar that should stay.

File: app/templates/signup.html [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+            {{ form.OHostName( size = 32)}}
+        </p>
+        <p>
+            {{ form.O_username.label }}
+            {{ form.O_username(size = 32)}}
+        </p>
+        <p>
+            {{ form.O_password.label }}
+            {{ form.O_password(size = 32)}}
+        </p>
```
Comment (haleyhagler227) [inline app/templates/signup.html]: username and password are the same for each

File: app/templates/signup.html [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+            {{ form.password.label }} <br>
+            {{ form.password(size=32) }}
+        </p>
+
+        <h2> OUTGOING MAIL SERVER </h2>
+        <p>
+            {{ form.OHostName.label }}
+            {{ form.OHostName( size = 32)}}
+        </p>
+        <p>
```
Comment (haleyhagler227) [inline app/templates/signup.html]: Add a OPortNum
```suggestion
        <p>
        {{ form.OPortNum.label }}
        {{ form.OPortNum(size = 32)}}
        </p>

```

File: app/templates/signup.html [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+            {{ form.IHostName(size = 32)}}
+        </p>
+        <p>
+            {{ form.username.label }}<br>
+            {{ form.username(size=32) }}
+        </p>
+        <p>
+            {{ form.password.label }} <br>
+            {{ form.password(size=32) }}
+        </p>
```
Comment (haleyhagler227) [inline app/templates/signup.html]: Add a IPortNum
```suggestion
        <p>
        {{ form.IPortNum.label }}
        {{ form.IPortNum(size = 32)}}
        </p>

```

File: app/templates/signup.html [hunk 7]
```diff
@@ -0,0 +1,54 @@
+{% extends "base.html" %}
+
```
Comment (haleyhagler227) [inline app/templates/signup.html]: Make a second file, named registerEmail.html, or anything kind of like that and put this content. Use Signup.html to get a username and a password (have two the password fields so they can be compared)

File: app/templates/signup.html [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+        
+        <p>
+            {{ form.name.label }}<br>
+            {{ form.name(size = 32)}}
+        </p>
+        <p>
+            {{ form.email.label }}<br>
+            {{ form.email(size = 32)}}
+        </p>
+
```
Comment (haleyhagler227) [inline app/templates/signup.html]: put the password field here. And only ask for it once

File: app/templates/signup.html [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+        }
+    </style>
+    <h1> Register </h1>
+    <form action="" method="post" novalidate>
+        {{form.hidden_tag() }}
+        
+        <p>
+            {{ form.name.label }}<br>
+            {{ form.name(size = 32)}}
+        </p>
```
Comment (haleyhagler227) [inline app/templates/signup.html]: I would remove this

File: app/templates/signup.html [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+
+{% block content %}
+    <div>
+    <style>
+    h2{
+        font-size: 8px;
+        color: grey;
+        }
+    </style>
+    <h1> Register </h1>
```
Comment (haleyhagler227) [inline app/templates/signup.html]: Register Email

**Review by haleyhagler227 — CHANGES_REQUESTED** [review]
Main point is signup looks like it is trying to play two functions: signing people up to the site and registering emails. I think it would be better to separate the two. So the email registration can be reused at other points in the app if necessary.

File: app/templates/signup.html (hunk 10 above)
Comment (jlc1541) [reply app/templates/signup.html]: Got you :)

**Review by jlc1541 — COMMENTED** [review]
I agree with you on this one. It would be better for the fluidity of the user experience to keep that nav bar. Nice Suggestion.

File: app/templates/signup.html (hunk 9 above)
Comment (jlc1541) [reply app/templates/signup.html]: That is probably best considering we already have a username.

File: app/templates/base.html (hunk 3 above)
Comment (jlc1541) [reply app/templates/base.html]: This makes a lot of sense to me. I've already seen this criticism so I understand that these changes are very significant to user experience.

**Pull request closed without merging**